liberty-maven-plugin/src/main/java/net/wasdev/wlp/maven/plugins/applications/InstallAppMojoSupport.java

```diff
@@ -154,8 +155,7 @@ private boolean containsJavaSource(MavenProject proj) {
     private boolean containsJavaSource(File dir) {
         File[] files = dir.listFiles();
         for (File file : files) {
-            if (file.isFile() && file.getName().lastIndexOf(".") != -1 
-                    && file.getName().substring(file.getName().lastIndexOf(".")).equalsIgnoreCase(".java")) {
+            if (file.isFile() && file.getName().lastIndexOf(".") != -1  && file.getName().toLowerCase().endsWith(".java")) {
```

Comment:
Not to belabor this but can't your remove file.getName().lastIndexOf(".") != -1 as well since you are checking for ".java"?